Validation/RecoTau/python/dataTypes/ValidateTausOnQCD_cff.py

```diff
@@ -25,8 +25,40 @@
 zttModifier = ApplyFunctionToSequence(zttLabeler)
 proc.TauValNumeratorAndDenominatorQCD.visit(zttModifier)
 
+#Set discriminators
+proc.RunHPSValidationQCD.discriminators = cms.VPSet(
```

Comment:
Just one quick question, will look into it in more detail tomorrow:
If I understand correctly, you overwrite here the "discriminators" already attached, keeping only the ones relevant for the process under consideration. Now I'm wondering if this can be done in a slightly nicer wary, e.g. by iterating over the discriminators already attached and removing those that don't have any of "DecayModeFinding", "CombinedIsolationDBSum", or "IsolationMVArun2" in their name. I don't know how easy/well-received it would be to come up with a solution like this, but it would reduce the number of modified lines by a lot.

Reply:
Hi @steggema, it is doable modifing Validation/RecoTau/src/TauTagValidation.cc. But I have found this way more elegant, instead of adding some if statements. Furthermore, with the if statements, we will have iso/muons/electrons discriminators for all the QCD/Ztt/Zmm/Zee cases.

Reply:
@aspiezia I probably didn't phrase my comment very clearly. I had something in mind along the lines of:

proc.RunHPSValidationQCD.discriminators = cms.VPSet([p for p in proc.RunHPSValidationQCD.discriminators if 'ByDecayModeFinding' in p.discriminator.value()])

which would save a lot of added lines.